File: Assignment_13/question3.c
////////////////////////////////////////////////////////////////////////////////
//
//  File Name   : question3.c
//  Description : Display pattern of capital letters row-wise.
//
//  Problem     : Accept number of rows and columns and display pattern.
//
//  Author      : Asawari Jadhav
//  Date        : 20/05/2025
//
////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>

void Pattern(int iRow, int iCol)
{

  int i = 0;
  int j = 0;
  char ch = '\0';

  for (i = 0; i < iRow; i++)
  {
    ch = 'A' + i;

    for (j = 0; j < iCol; j++)
    {
      printf("%c\t", ch);
    }
    printf("\n");
  }
}

int main()
{
  int iValue1 = 0, iValue2 = 0;

  printf("Enter number of rows and columns\n");
  scanf("%d %d", &iValue1, &iValue2);

  Pattern(iValue1, iValue2);

  return 0;
}

File: Assignment_50/question3.java
//////////////////////////////////////////////////////////////////////////////////////////////
//
//  File Name     : question3.java
//  Description   : Accept number of rows and columns from user and display hollow star pattern
//  Problem       : Display * on borders and diagonals, spaces inside
//
//  Author        : Asawari Jadhav
//
//  Date          : 21-08-2025            
//
//////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern 
{
    public void DisplayPattern(int iRow, int iCol) 
    {
        for (int i = 0; i < iRow; i++) 
        {
            for (int j = 0; j < iCol; j++) 
            {
                if (i == 0 || i == iRow - 1 || j == 0 || j == iCol - 1 || i == j || i + j == iCol - 1)
                {
                    System.out.print("* ");
                }
                    
                else
                {
                    System.out.print("  ");
                }
                    
            }
            System.out.println();
        }
    }
}

public class question3 
{
    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number of rows: ");
        int iRow = sc.nextInt();
        System.out.print("Enter number of columns: ");
        int iCol = sc.nextInt();

        Pattern pobj = new Pattern();
        pobj.DisplayPattern(iRow, iCol);

        
    }
}
